hydra/controllers/client.py

```diff
@@ -816,3 +831,78 @@ def disable_metrics(self):
 
         self.restart_service()
         self.app.log.info(f'Successfully disabled all metrics reporting.')
+
+    @ex(
+        arguments=[
+            (
+                ['-n', '--name'],
+                {
+                    'help': 'name of network to join',
+                    'action': 'store',
+                    'dest': 'name'
+                }
+            ),
+            (
+                ['-v', '--version'],
+                {
+                    'help': 'version of binary',
+                    'action': 'store',
+                    'dest': 'version'
+                }
+            ),
+            (
+                ['-D', '--destroy'],
+                {
+                    'help': 'destroy existing directory',
+                    'action': 'store_true',
+                    'dest': 'destroy'
+                }
+            ),
+            (
+                ['-d', '--destination'],
+                {
+                    'help': 'destination directory',
+                    'action': 'store',
+                    'dest': 'destination'
+                }
+            ),
+        ]
+    )
+    def upgrade_binary(self, version=None, destroy=False):
+        name = self.app.utils.env_or_arg(
+            'name', 'HYDRA_NETWORK', or_path='.hydra_network', required=True)
+
+        destination = self.app.pargs.destination or self.app.utils.path(name)
+
+        service_name = f'{name}.service'
+        systemd_service = f'/etc/systemd/system/{service_name}'
+
+        if os.path.exists(systemd_service):
+            self.stop_service()
+        else:
+            self.app.log.info(f'Service not installed.  Attempting to stop executable.')
+            self.app.client.find_and_kill_executable(destination)
+
+        if os.path.exists(destination):
+            if not self.app.pargs.destroy:
+                self.app.log.error(f'Directory exists, use -D to delete: {destination}')
+                return
+            rmtree(destination)
```

Comment:
We definitely don't want to be deleting or recreating the network folder for a binary upgrade

Reply:
Agreed!